src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            Book book = null;
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                String title = resultSet.getString(TITLE);
+                BigDecimal price = resultSet.getBigDecimal(PRICE);
+
+                book = new Book(id, title, price);
+            }
+            return Optional.ofNullable(book);
+        } catch (SQLException e) {
+            throw new RuntimeException(DB_CONNECTION_ERROR_MESSAGE, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            List<Book> bookList = new ArrayList<>();
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                Long id = resultSet.getObject(ID, Long.class);
+                String title = resultSet.getString(TITLE);
+                BigDecimal price = resultSet.getBigDecimal(PRICE);
+                bookList.add(new Book(id, title, price));
+            }
+            return bookList;
+        } catch (SQLException e) {
+            throw new RuntimeException(DB_CONNECTION_ERROR_MESSAGE, e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            if (book.getId() == null) {
+                throw new RuntimeException(NULL_ID_ERROR_MESSAGE);
+            }
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 0) {
+                throw new RuntimeException(UPDATE_ERROR_MESSAGE + book.getId());
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new RuntimeException(DB_CONNECTION_ERROR_MESSAGE, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 0) {
+                throw new RuntimeException(DELETE_ERROR_MESSAGE + id);
```

Comment:
Why do you need to throw exception, if your method returns boolean value of whether a book was deleted or not. You are now always returning `true` or throwing an exception which makes no sense. Better don't throw an exception here, but simply return something like `return statement.executeUpdate() > 0`